2-mandatory/3-groceries.js

```diff
@@ -26,7 +26,33 @@ Exercise 1:
   The weeklyGroceriesToBuy array shouldn't contain any repeating items.
 */
 // Gather all week item names into this array
+//*********************************************************
+// A function to return unique items
+//*********************************************************
+function uniqueItems(inputArray) {
+  let uniqueChars = [];
+  inputArray.forEach((c) => {
+     if (!uniqueChars.includes(c)) {
+        uniqueChars.push(c);
+     }
+
+  });
+  return uniqueChars;
+}
 let weeklyGroceriesToBuy = [];
+//weeklyGroceriesToBuy.push();
+
+keyNames = Object.keys(weeklyMealPlan);
+valueNames = Object.values(weeklyMealPlan);
+for (let i = 0; i < keyNames.length; i++) {
```

Comment:
It is quite difficult to understand this code as you use variable with the names keyNames and keyValues and the use of multi-dimensional arrays (valueNames[i][j]) .

Also, you do not need the uniqueItems function. Your solution loops through the ingredients twice. Why not check if it is unque in the first loop.

The code could be much simpler to read, and more efficient:

for (let day in weeklyMealPlan) {
  let ingredients = weeklyMealPlan[day];
  ingredients.forEach((ingredient) => {
    if (!weeklyGroceriesToBuy.includes(ingredient)) {
      weeklyGroceriesToBuy.push(ingredient);
    }
  });
}